gammapy/datasets/simulate.py

```diff
@@ -574,3 +574,146 @@ def run(self, dataset, observation=None):
+
+    Parameters
+    ----------
+    energy_axis_true : `~gammapy.maps.MapAxis`
+        True energy axis.
+    energy_axis : `~gammapy.maps.MapAxis`
+        Reconstructed energy axis.
+    spatial_width : float or `~astropy.units.Quantity`
+        Spatial window size. A float is assumed to be in degree.
+    spatial_spatial_bin_size : float or `~astropy.units.Quantity`, optional
+        Pixel size used for sampling. A float is assumed to be in degree.
+        Default is 0.01 degree.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}, optional
+        Defines random number generator initialisation via the `~gammapy.utils.random.get_random_state` function.
+    oversample_energy_factor : int, optional
+        Defines an oversampling factor for the energies; it is used only when sampling
+        an energy-dependent time-varying source.
+    t_delta : `~astropy.units.Quantity`, optional
+        Time interval used to sample the time-dependent source.
+    keep_mc_id : bool, optional
+        Flag to tag sampled events from a given model with a Montecarlo identifier.
+        Default is True. If set to False, no identifier will be assigned.
+    """
+
+    def __init__(
+        self,
+        energy_axis_true,
+        energy_axis,
+        spatial_width,
+        spatial_bin_size=0.01 * u.deg,
+        random_state="random-seed",
+        oversample_energy_factor=10,
+        t_delta=0.5 * u.s,
+        keep_mc_id=True,
+    ):
+        self.random_state = get_random_state(random_state)
+        self.energy_axis_true = energy_axis_true
+        self.energy_axis = energy_axis
+        self.spatial_width = u.Quantity(spatial_width, "deg")
+        self.spatial_bin_size = u.Quantity(spatial_bin_size, "deg")
+        self.oversample_energy_factor = oversample_energy_factor
+        self.t_delta = t_delta
+        self.keep_mc_id = keep_mc_id
+
+    def _create_dataset(self, observation, models=None, dataset_name=None):
+        """create dataest used for sampling"""
+
+        # import here to prevent circular import
+        from gammapy.makers import MapDatasetMaker
+
+        if models is None:
+            models = Models()
+
+        if dataset_name is None:
+            dataset_name = "simulated-dataset"
+
+        if not np.any(
+            [
+                isinstance(m, FoVBackgroundModel) and m.dataset_name == dataset_name
+                for m in models
+            ]
+        ):
+            models.append(FoVBackgroundModel(dataset_name=dataset_name))
+
+        if observation.pointing.mode is not PointingMode.POINTING:
+            raise NotImplementedError(
+                "Only observations with fixed pointing in ICRS are supported"
+            )
+        pointing_icrs = observation.pointing.fixed_icrs
+        geom = WcsGeom.create(
+            skydir=pointing_icrs,
+            width=(
+                self.spatial_width.to_value(u.deg),
```

Comment:
You don't need to convert to degree here. `WcsGeom.create` accepts quantities too.